ledger/eval/txntracer.go

```diff
@@ -48,12 +48,50 @@ type StateDeltaSubset struct {
 }
 
 func convertStateDelta(delta ledgercore.StateDelta) StateDeltaSubset {
+	// The StateDelta object returned through the EvalTracer has its values deleted between txn groups to avoid
+	// reallocation during evaluation.
+	// This means the map values need to be copied (to avoid deletion) since they are all passed by reference.
+	kvmods := make(map[string]ledgercore.KvValueDelta)
```

Comment:
nit: I'd recommend instantiating the maps with the right capacity, since you know what it is.

Reply:
+1 to Jason